dll/win32/shell32/folders/CRecycleBin.cpp

```diff
@@ -473,22 +462,93 @@ HRESULT WINAPI CRecycleBinItemContextMenu::QueryContextMenu(HMENU hMenu, UINT in
     return idHigh ? MAKE_HRESULT(SEVERITY_SUCCESS, 0, idHigh - idCmdFirst + 1) : S_OK;
 }
 
-static BOOL ConfirmDelete(LPCMINVOKECOMMANDINFO lpcmi, UINT cidl, LPCITEMIDLIST pidl, const BBITEMDATA &Data)
+static BOOL ConfirmDelete(LPCMINVOKECOMMANDINFO lpcmi, UINT cidl, LPCITEMIDLIST pidl)
 {
+    BBITEMDATA *pData;
     if (lpcmi->fMask & CMIC_MASK_FLAG_NO_UI)
     {
         return TRUE;
     }
-    else if (cidl == 1)
+    else if (cidl == 1 && (pData = ValidateItem(pidl)) != NULL)
     {
         const UINT ask = IsFolder(pidl) ? ASK_DELETE_FOLDER : ASK_DELETE_FILE;
-        return SHELL_ConfirmYesNoW(lpcmi->hwnd, ask, GetItemOriginalFileName(Data));
+        return SHELL_ConfirmYesNoW(lpcmi->hwnd, ask, GetItemOriginalFileName(*pData));
     }
-    WCHAR buf[MAX_PATH];
+    WCHAR buf[42];
```

Comment:
Why 42 ?

Reply:
Mainly to avoid using MAX_PATH for something that is not a path. 42 is a nice magic number and plenty for storing a single integer.

Reply:
OK. However, I would suggest adding a comment for that "magic number".